### PR TITLE
Adding basic lobby owner structure

### LINKED ISSUE #108: [Feature Request]: Lobby Owner
### What should be implemented?

If a user creates a lobby, they are the lobby owner. If the owner leaves their lobby, a random user becomes the owner. In the future, the lobby owner could have more rights like changing settings and deleting the lobby (see #63 )

### PR DESCRIPTION
**Pull request by JayPi4c**
## What type of PR is this? (keep all applicable)

- Feature

## Description

Adding basic lobby owner functionality. The `ILobby` interface now holds a `owner` variable. The client may now show different views according to the lobby member status. Lobby owners are displayed with a `(Owner)` suffix to their names in the `lobbyUsersList`. If the owner ever leaves the lobby, another member of the lobby is selected to inherit the role. The `UserLeftLobbyMessage` has been updated to transport a whole lobby instead of the single user who left the lobby. Alternatively, it would have been possible to simply send the current lobby owner next to the user who left the lobby in order to update any possible changes in the roles. Since this action (a user leaving the lobby) is the only one that allows for a change in the lobby owner role, it is the only message that has been updated. 

## Related Tickets & Documents

- Closes #108

### CODE REVIEW

File: client/src/main/java/org/schlunzis/kurtama/client/service/impl/LobbyService.java [hunk 1]
```diff
@@ -59,7 +59,11 @@ public void leaveLobby() {
 
     @EventListener
     void onUserLeftLobbyMessage(UserLeftLobbyMessage ullm) {
-        Platform.runLater(() -> lobbyUsersList.remove(ullm.leavingLobbyMember().getUsername()));
+        Platform.runLater(() -> {
+            lobbyUsersList.clear();
+            ullm.lobby().getUsers().forEach(u ->
+                    lobbyUsersList.add(u.getUsername() + (u.equals(ullm.lobby().getOwner()) ? " (Owner)" : "")));
```
Comment (Til7701) [inline client/src/main/java/org/schlunzis/kurtama/client/service/impl/LobbyService.java]: A list item factory would be cleaner.

**Review by Til7701 — CHANGES_REQUESTED** [review]
Bug:
1. Create Lobby with test1
2. Join with test2
3. Leave with test1
4. test2 cannot leave, since the lobby no longer exists

File: client/src/main/java/org/schlunzis/kurtama/client/service/impl/LobbyService.java (hunk 1 above)
Comment (JayPi4c) [reply client/src/main/java/org/schlunzis/kurtama/client/service/impl/LobbyService.java]: You mean something like changing the `lobbyUsersList` to contain Users and in the controller we set a cell factory for the `usersListView`? Comparable to the `lobbyListView` in the main menu.

**Comment by JayPi4c** [comment]
The bug is even more basic. Currently no last user can leave the lobby. In the serverside `LobbyService` in th method `onLeaveLobbyRequest(...)` we can't obtain the lobby after the last member has left. So we fail to construct the `UserLeftLobbyMessage` which leads to a failed leave lobby action... A simple solution would be to wrap the problematic code in a `try-catch` block. But this feels kinda wrong...
```java
try {
    ServerLobby lobby = lobbyManagement.getLobby(request.lobbyID());
    var userLeftLobbyMessage = new UserLeftLobbyMessage(lobby.toDTO());
    informUsersInLobby(userLeftLobbyMessage, request.lobbyID(), cmc);
} catch (LobbyNotFoundException e) {
    log.debug("Lobby not found. Probably it's been removed.");
}
```

Any suggestions?

File: client/src/main/java/org/schlunzis/kurtama/client/service/impl/LobbyService.java (hunk 1 above)
Comment (Til7701) [reply client/src/main/java/org/schlunzis/kurtama/client/service/impl/LobbyService.java]: Correct

**Comment by Til7701** [comment]
`LobbyManagement.leaveLobby()` could to one of the following.

1. return the lobby
2. return a status enum
3. return a boolean indicating whether or not the lobby still exists

**Comment by JayPi4c** [comment]
I think returning the lobby or creating special enums for this problem are not the best choice. Therefore I chose the latter option.

**Review by Til7701 — APPROVED** [review]

**Pull request merged**